src/gd_avif.c

```diff
@@ -0,0 +1,667 @@
+/**
+ * File: AVIF IO
+ *
+ * Read and write AVIF images using libavif (https://github.com/AOMediaCodec/libavif) .
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <limits.h>
+#include <math.h>
+
+#include "gd.h"
+#include "gd_errors.h"
+#include "gdhelpers.h"
+
+#ifdef HAVE_LIBAVIF
+#include <avif/avif.h>
+#include "internal.h"
+// internal.h is only used for AVIF_MIN() and AVIF_CLAMP() macros
+
+/*
+  Define defaults for encoding images:
+    DEFAULT_CHROMA_SUBSAMPLING: 4:2:0 is commonly used for Chroma subsampling.
+    DEFAULT_MIN_QUANTIZER, DEFAULT_MAX_QUANTIZER: 
+      We need more testing to really know what quantizer settings are optimal,
+      but teams at Google have been using maximum=30 as a starting point.
+    DEFAULT_QUALITY: following gd conventions, -1 indicates the default.
+    DEFAULT_SPEED: AVIF_SPEED_DEFAULT is -1. This simply tells the AVIF encoder to use the default speed.
+*/
+
+#define DEFAULT_CHROMA_SUBSAMPLING AVIF_PIXEL_FORMAT_YUV420
+#define DEFAULT_QUANTIZER 30
+#define DEFAULT_QUALITY -1
+#define DEFAULT_SPEED AVIF_SPEED_DEFAULT
+
+// This initial size for the gdIOCtx is standard among GD image conversion functions
+#define NEW_DYNAMIC_CTX_SIZE 2048
+
+// Our quality param ranges from 0 to 100.
+// To calculate quality, we convert from AVIF's quantizer scale, which runs from 63 to 0.
+#define MAX_QUALITY 100
+
+// For computing the number of tiles and threads to use during encoding
+// Maximum threads are from libavif/contrib/gkd-pixbuf/loader.c
+#define MIN_TILE_AREA (512 * 512)
+#define MAX_TILES 6
```

Comment:
MAX_TILES should be a power of 2 because the number of tile rows and the number of tile columns must be powers of 2.

Reply:
Yes! I'd noticed that one and changed it to 8 last night. Looks like the new code isn't showing up in this view :)

Reply:
Ben: I am sorry I commented on old issues that have been fixed. (I responded to your questions for me, which were posted on an old view of the PR, and naturally looked at related code.)

Reply:
normally I wouldn't manage to spot these two issues so quickly :)